grafana/grafana
# Elasticsearch: Removed reference to obsolete esVersion value

in a previous PR ( https://github.com/grafana/grafana/pull/63341 )  we made the "elastic database version" auto-detected by grafana.

previously the user had to specify the elasticsearch version on the datasource config page.

this PR removes all the now-unnecessary fields, and values, that referred to the old functionality, also updates the documentation.

the diff is large, but most of it is updates to the tests, and deleting code from non-tests.

how to test:
1. try out the elastic datasource config page, try to create a new elastic datasource, modify it's attributes, it should work.
